node/src/actors/chain_manager/mining.rs

```diff
@@ -291,13 +288,14 @@ impl ChainManager {
         let current_retrieval_count = Arc::new(AtomicU16::new(0u16));
         let maximum_retrieval_count = self.data_request_max_retrievals_per_epoch;
 
-        for (dr_pointer, dr_state) in dr_pointers.into_iter().filter_map(|dr_pointer| {
-            // Filter data requests that are not in data_request_pool
-            self.chain_state
-                .data_request_pool
-                .data_request_state(&dr_pointer)
-                .map(|dr_state| (dr_pointer, dr_state.clone()))
-        }) {
+        for (dr_pointer, dr_state) in self
+            .chain_state
+            .data_request_pool
+            .data_request_pool
+            .iter()
+            // Clone the dr_state because it must be valid for a static lifetime
+            .map(|(dr_pointer, dr_state)| (*dr_pointer, dr_state.clone()))
+        {
```

Comment:
The problem here if that requests that are in Reveal and Tally stage appear here as available but we only want requests in Commit stage

Reply:
Good catch, so we still need to keep track of all the data requests in commit stage. So instead of a `BTreeMap<Epoch, HashSet<Hash>>` we can use a `HashSet<Hash>`, or we can use `filter_map` to iterate over all the requests and check the stage.